### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by rotimiazeez**
# OOP school library: Preserve Data🎉 

> In this project you will preserve the data from your app and load preserved data on startup💪🏽.

Additional description of the project and its features ➕:

- Check the documentation for Ruby's JSON library.
- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)
- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

## Built With ❤️ 

- Ruby

## Deployed Preview 🚀 
[Deployed Preview](Coming soon)

### CODE REVIEW

File: people.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"json_class":"Student","id":null,"name":null,"age":{"name":"Rahyan","age":"12","parent_permission":false},"parent_permission":null}]
```
Comment (arslanbisharat) [inline people.json]: - [x] This is an example of `people.json`:
`[{"json_class":"Student","name":"Sam","age":"14","parent_permission":true},{"json_class":"Student","name":"Pom","age":"16","parent_permission":false},{"json_class":"Teacher","name":"Fox","age":"56","parent_permission":true,"specialization":"Starships"},{"json_class":"Student","name":"Poppy","age":"20","parent_permission":false}]`

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: rental.rb [hunk 2]
```diff
@@ -15,4 +15,13 @@ def initialize(date, person, book)
...
   end
+
+  def to_json(*args)
+    {
+      JSON.create_id => self.class.name,
+      'date' => @date,
+      'book_title' => @book.title,
+      'person_id' => @person.id
+    }.to_json(*args)
+  end
```
Comment (mikethreels) [inline rental.rb:26]: - [ ] [OPTIONAL] Sorry that it wasn't pointed out earlier and therefore this is an OPTIONAL suggestion, however when creating some teachers, students, books and rentals the application works how it is supposed to however when closing the application I get an error and I'm not able to restart it anymore. see below the print screen of the error.
Przechwytywanie

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete! it's time to merge it 🚀 :
Congratulations! 🎉 

- [x] [OPTIONAL]  sorry that it wasn't pointed out earlier, however, there is still an issue in your application that renders it unusable after closing it, although this is not a requirement for you to move on to the next project I highly recommend you to fix this before you do because the application is not working how it is supposed to.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**